aten/src/ATen/native/quantized/cpu/serialization_versions.h

```diff
@@ -0,0 +1,164 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/core/List.h>
+
+#include <ATen/native/quantized/cpu/fbgemm_utils.h>
+#include <ATen/native/quantized/cpu/qnnpack_utils.h>
+
+#include <tuple>
+
+using SerializationType = std::tuple<
+  std::string /*name of the op that is serialized*/,
+  int64_t /*version*/,
+  std::vector<at::Tensor> /*list of all non-optional tensors*/,
+  std::vector<c10::optional<at::Tensor>> /*list of all optional tensors*/,
+  std::vector<double> /*list of all doubles*/,
+  std::vector<int64_t> /*list of all integers*/
+>;
+
+/* Legacy serialization types */
+
+
+/* Convolution prepacked parameters serialization.
+ *
+ * Files that need to be updated if version changes:
+ * - ATen/native/quantized/cpu/fbgemm_utils.cpp
+ * - jit/passes/onnx/unpack_quantized_weights.cpp
+ *
+ * Version 1 (Legacy)
+ * - Fields:
+ *  1. weight
+ *  2. bias
+ *  3. stride x kSpatialDim
+ *  4. padding x kSpatialDim
+ *  5. dilation x kSpatialDim
+ *  6. groups
+ *
+ * Version 2+
+ * - Fields:
+ *  1. name of the op ("conv")
+ *  2. version number (int64_t)
+ *  3. list of all non-optional tensors (vector<Tensor>)
+ *    1. unpacked weight
+ *  4. list of all optional tensors (vector<optional<Tensor>>)
+ *    1. bias
+ *  5. list of all doubles
+ *    EMPTY
+ *  6. list of all integers (vector<int64_t> of size (1 + 4 * kSpatialDim + 1))
```

Comment:
I count 3 * kSpatialDim

Reply:
Oh this bled through from later PRs -- it is supposed to be 1 + 3 * kSpatialDim. In the next PR it is changed to 1 + 4 kSpatialDim + 1 => groups (1) + strides/padding/dilation/output_padding(kSpatialDim) + transpose